webClient/src/app/editor/code-editor/monaco/monaco.component.ts

```diff
@@ -231,6 +250,37 @@ export class MonacoComponent implements OnInit, OnChanges {
     });
   }
 
+  copyPermalink(event: any){
+      const lines = event.target.position.lineNumber;
+      const activeFile = this.editorControl.fetchActiveFile();
+      let filePath = '';
+      let link = '';
+      if(activeFile.model.isDataset){
+        filePath = activeFile.model.path;
+        link = `${window.location.origin}${window.location.pathname}?pluginId=${this.pluginDefinition.getBasePlugin().getIdentifier()}:data:{"type":"openDataset","name":"${encodeURIComponent(filePath)}","lines":"${lines}","toggleTree":true}`;
+      } else {
+        filePath = activeFile.model.path + "/" + activeFile.model.name;
+        link = `${window.location.origin}${window.location.pathname}?pluginId=${this.pluginDefinition.getBasePlugin().getIdentifier()}:data:{"type":"openFile","name":"${encodeURIComponent(filePath)}","lines":"${lines}","toggleTree":true}`;
+      }
+      navigator.clipboard.writeText(link).then(() => {
+        this.log.debug("Link copied to clipboard");
+        this.snackBar.open("Copied link successfully", 'Dismiss', { duration: MessageDuration.Short, panelClass: 'center' });
+      }).catch((error) => {
+        console.error("Failed to copy link to clipboard"+ error);
+      });
+  }
+
+  copyLine(event: any){
+      const lines = event.target.position.lineNumber;
+      const lineContent = this.editor.getModel().getLineContent(lines);
+      navigator.clipboard.writeText(lineContent).then(() => {
+        this.log.debug("Line copied to clipboard");
+        this.snackBar.open("Copied line successfully", 'Dismiss', { duration: MessageDuration.Short, panelClass: 'center' });
```

Comment:
We probably don't need this snackbar. It's rare I see a pop-up for mouse copy actions